pkg/git/git.go

```diff
@@ -117,9 +150,18 @@ func (g *CLIGit) CreateBranch() error {
 	return nil
 }
 
-// CreateRepository bootstraps a plain repository at a given location.
-func (g *CLIGit) CreateRepository() error {
-	return errors.New("implement me")
+func (g *CLIGit) CreateNewBranch(branch string) error {
+	args := []string{
+		"--git-dir", filepath.Join(g.Directory, ".git"),
+		"--work-tree", g.Directory,
+		"checkout",
+		"-b",
+		branch,
+	}
+	if err := g.runGitCmd(args...); err != nil {
+		return fmt.Errorf("failed to create new branch %s: %w", branch, err)
+	}
+	return nil
```

Comment:
I'm not quite follow what's happening here. :D Did you create a new function which does the same as the one above it? :) The CreateBranch check whether the selected branch is the one that is already checked out in which case it doesn't create it.

Reply:
I checked it out. Yeah, I think you can use create branch and modify it as you do with CreateNewBranch and then delete CreateNewBranch. :)

Reply:
the existing `CreateBranch` uses the `g.Branch`, where as I want to take in an argument :thinking:

Reply:
Yeah, I saw. Just modify it to take one. :) I think that's fine, rather than two separate functions which do the same thing.

Reply:
:+1:

Reply:
fixed